runtime/Java/src/org/antlr/v4/runtime/BufferedTokenStream.java

```diff
@@ -441,15 +441,17 @@ protected int previousTokenOnChannel(int i, int channel) {
 
 	@Override
 	public String getText() {
+		fill();
 		return getText(Interval.of(0,size()-1));
 	}
 
 	@Override
 	public String getText(Interval interval) {
+		lazyInit();
 		int start = interval.a;
 		int stop = interval.b;
 		if ( start<0 || stop<0 ) return "";
-		fill();
+		sync(stop);
```

Comment:
📝 This is the only part of the change that really works. I'd keep this line and remove the others.